app/controllers/articles_controller.rb

```diff
@@ -6,4 +6,29 @@ def index
     def show
         @article = Article.find(params[:id])
     end
+
+    def new
+        @article = Article.new
+        @categories = Category.all
+    end
+
+    def create
+        @article = Article.new(article_params)
+        @article.update(category: find_category)
+        if @article.save
+            redirect_to article_path(@article)
+        else
+            redirect_to new_article_path, notice: "Every field needs to be filled in!"
```

Comment:
This is not a blocker, but you know that AR models have error messages for each attribute that has a validation? Those error messages can be passed into the view layer.

Reply:
We know this way of showing error messages, from the rails documentation:
https://guides.rubyonrails.org/active_record_validations.html#displaying-validation-errors-in-views
```

<% if @article.errors.any? %> 
 <div id="error_explanation">
    <h2><%= pluralize(@article.errors.count, "error") %>
 prohibited this article from being saved:</h2>     
<ul>    
<% @article.errors.full_messages.each do \|msg\| %>
      <li><%= msg %></li>    
<% end %>    
</ul>  
</div>
<% end %>
--
```

But for some reason it was not working for us, Oliver suggested this solution.

Reply:
I would NOT do this in the view, but rather in the controller. But as I said, it's not a blocker. I feel compelled to comment on another thing though: "for some reason" is not good enough of an explanation.